### PR TITLE
reverse web proxy addition

### PR DESCRIPTION
**Pull request by shubhamjain0594**
* Added nginx and gunicorn to support as reverse web proxy and thus giving access to raqmn through port 80, which is running on gunicorn on 8000
* Modified raqmn_data app to use raqmn_api as library rather than project
* Modified drivers to accommodate changes of raqmn and serve all interaction with server through raqmn, including posting to giles server

### CODE REVIEW

File: air_quality_monitoring/raqmn/raqmn_api/views.py [hunk 1]
```diff
@@ -76,4 +68,35 @@ def get_average_data_12_hrs(request, uuid):
...
+	Returns the data of all sensors
+	"""
+	query_url = 'http://localhost:8079/api/query'
+	query = "select *"
+	r = requests.post(query_url, query)
+	return r.content
+
+def get_data_by_uuid_helper(uuid):
+	"""
+	Returns the data of sensor specified by uuid of last 30 days
```
Comment (shankari) [inline air_quality_monitoring/raqmn/raqmn_api/views.py]: This is really "Returns the data from the last 30 days for the sensor that is specified by the input UUID.

File: air_quality_monitoring/raqmn/raqmn_data/templates/raqmn_data/plot.html [hunk 2]
```diff
@@ -35,8 +35,8 @@
 	{% endfor %}
 	var graph = new Rickshaw.Graph({
```
Comment (shankari) [inline air_quality_monitoring/raqmn/raqmn_data/templates/raqmn_data/plot.html]: Does this mean that we are now using rickshaw for the plotting? I don't recall the original change for this...

File: air_quality_monitoring/raqmn_nginx.conf [hunk 3]
```diff
@@ -0,0 +1,23 @@
+server {
+  listen 80; 
+  client_max_body_size 4G;
+  keepalive_requests 100000;
+  keepalive_timeout 3000;
+  add_header Accept-Ranges bytes;
+  server_name 127.0.0.1 localhost;
+  location /static/ {
+    alias /home/dora/Study/BTP_Bhaskaran/raqmn/utils/Real-Time-Air-Quality-Sensoring-Network/air_quality_monitoring/staticfiles/;
```
Comment (shankari) [inline air_quality_monitoring/raqmn_nginx.conf]: This looks like a path from your laptop...

File: smap/sources/conf/safar_conf.ini [hunk 4]
```diff
@@ -5,7 +5,7 @@ Port = 8003
 uuid = 6fa21782-27b2-11e2-9a49-370ecdac5e02
 
 [report 0]
-ReportDeliveryLocation = http://localhost:8079/add/MYAPIKEY
+ReportDeliveryLocation = http://localhost/raqmn_api/add/6fa21782-27b2-11e2-9a49-370ecdac5e02/
```
Comment (shankari) [inline smap/sources/conf/safar_conf.ini]: Again, key should not be hardcoded?

**Comment by shankari** [comment]
At a high level, I am not sure what you are trying to do here. Are you trying to have a server on port 80 that redirects to port 8080 or something? If so, why do you need to have both nginx and gunicorn? Isn't one sufficient?

File: air_quality_monitoring/raqmn/raqmn_api/views.py (hunk 1 above)
Comment (shubhamjain0594) [inline air_quality_monitoring/raqmn/raqmn_api/views.py]: Yes it is a helper function that returns data of last 30 days for the sensor specified by uuid. As we proceed I will modify this API to get data of sensor specified by uuid and for time given as input.

File: air_quality_monitoring/raqmn/raqmn_data/templates/raqmn_data/plot.html (hunk 2 above)
Comment (shubhamjain0594) [inline air_quality_monitoring/raqmn/raqmn_data/templates/raqmn_data/plot.html]: This is a beginning towards trying out other Graph APIs and raqmn_data App will serve as front end of our Project. So this is first step towards it.

File: smap/sources/conf/safar_conf.ini (hunk 4 above)
Comment (shubhamjain0594) [inline smap/sources/conf/safar_conf.ini]: Right now, we don't have any checks and this key is an example of how keys look like, this is same as uuid but as we proceed every sensor will have to register and get a unique uuid for itself which will also be used as key. This is just for an example. Actual keys wont be released.

**Comment by shubhamjain0594** [comment]
Django servers are not meant for production. Hence we deploy them using nginx and gunicorn, nginx diverts the request at port 80 to port where django app is running. Gunicorn is used for deployment of django app. 
http://serverfault.com/questions/331256/why-do-i-need-nginx-and-something-like-gunicorn
The answer above provides a much better understanding of why nginx-gunicorn-django structure.

File: air_quality_monitoring/raqmn_nginx.conf (hunk 3 above)
Comment (shubhamjain0594) [inline air_quality_monitoring/raqmn_nginx.conf]: Solved in latest commit

File: air_quality_monitoring/raqmn/raqmn_api/views.py (hunk 1 above)
Comment (shankari) [inline air_quality_monitoring/raqmn/raqmn_api/views.py]: Right, so please update the documentation to reflect this to reduce confusion in the future.

File: air_quality_monitoring/raqmn/raqmn_data/templates/raqmn_data/plot.html (hunk 2 above)
Comment (shankari) [inline air_quality_monitoring/raqmn/raqmn_data/templates/raqmn_data/plot.html]: I understood that this is what it was doing, but I don't see a lot of discussion/documentation on this option. Can you add some comments on the architecture here and how you are querying the timeseries database?

**Comment by shankari** [comment]
Once you improve the docs, I will merge!

**Comment by shubhamjain0594** [comment]
Docs changes added.

**Pull request merged**